src/lxc/attach.c

```diff
@@ -1384,11 +1384,14 @@ int lxc_attach(struct lxc_container *container, lxc_attach_exec_t exec_function,
 
 	if (pid == 0) {
 		if (options->attach_flags & LXC_ATTACH_TERMINAL) {
-			ret = pthread_sigmask(SIG_SETMASK,
-					      &terminal.tty_state->oldmask, NULL);
-			if (ret < 0) {
-				SYSERROR("Failed to reset signal mask");
-				_exit(EXIT_FAILURE);
+			if (terminal.tty_state)
+			{
+				ret = pthread_sigmask(SIG_SETMASK,
+							  &terminal.tty_state->oldmask, NULL);
+				if (ret < 0) {
+					SYSERROR("Failed to reset signal mask");
+					_exit(EXIT_FAILURE);
+				}
```

Comment:
I think we should not open-code this. Can you add sm like the following on top of your patch, please?

file:///home/brauner/src/git/lxc/src/lxc/tmp/0001-terminal-introduce-lxc_terminal_signal_sigmask_safe_.patch
file:///home/brauner/src/git/lxc/src/lxc/tmp/0002-attach-use-lxc_terminal_signal_sigmask_safe_blocked.patch

Reply:
I gave these patches a test with in my setup, no issues seen.